Sami/Parser/ClassVisitor/PropertyClassVisitor.php

```diff
@@ -50,58 +49,26 @@ public function visit(ClassReflection $class)
      * Adds a new property to the class using an array of tokens.
      *
      * @param ClassReflection $class       Class reflection
-     * @param string          $propertyTag Property tag contents
+     * @param string|array    $propertyTag Property tag contents
      *
      * @return bool
      */
     protected function injectProperty(ClassReflection $class, $propertyTag)
     {
-        if (!$data = $this->parseProperty($propertyTag)) {
-            return false;
-        }
-
-        $property = new PropertyReflection($data['name'], $class->getLine());
-        $property->setDocComment($data['description']);
-        $property->setShortDesc($data['description']);
-
-        if (isset($data['hint'])) {
-            $property->setHint(array(array($data['hint'], null)));
-        }
-
-        $class->addProperty($property);
-
-        return true;
-    }
-
-    /**
-     * Parses the parts of an @property tag into an associative array.
-     *
-     * @param string $tag Property tag contents
-     *
-     * @return array
-     */
-    protected function parseProperty($tag)
-    {
-        // Account for default array syntax
-        $tag = str_replace('array()', 'array', $tag);
-
-        /** @var PropertyTag $propertyTag */
-        $propertyTag = $this->context->getDocBlockParser()->getTag('@property '.$tag);
-        $propertyName = $propertyTag->getVariableName();
+        if (is_array($propertyTag) && count($propertyTag) == 3 && !empty($propertyTag[1])) {
```

Comment:
* the `is_array` check isn't needed if you add `array` type hint to the `$propertyTag` parameter
* for `@property type short description` you have `$propertyTag` from 4 elements and that in your code version would result in `description` word being lost and you'll have just `short` store as property description